src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -364,9 +364,18 @@ bool ColorControlServer::computeNewColor16uValue(ColorControlServer::Color16uTra
     }
     else if (p->finalValue > p->initialValue)
     {
-        newValue32u = ((uint32_t)(p->finalValue - p->initialValue));
-        newValue32u *= ((uint32_t)(p->stepsRemaining));
-        newValue32u /= ((uint32_t)(p->stepsTotal));
+        newValue32u = static_cast<uint32_t>(p->finalValue - p->initialValue);
+        newValue32u *= static_cast<uint32_t>(p->stepsRemaining);
+
+        /*
+            stepsTotal should always be at least 1,
+            still, prevent division by 0 and skips a meaningless division by 1
+        */
+        if (p->stepsTotal > 1)
```

Comment:
stepsTotal should always be at least 1, should we change to ` if (p->stepsTotal >= 1)`

Reply:
If stepsTotal is 1 then the division by 1 isn't really adding anything, that is why is excluded 1 also.